resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodsSnapshot.scala

```diff
@@ -65,14 +72,33 @@ object ExecutorPodsSnapshot extends Logging {
         case "pending" =>
           PodPending(pod)
         case "running" =>
+          // If we're checking all containers look for any non-zero exits
           if (shouldCheckAllContainers &&
             "Never" == pod.getSpec.getRestartPolicy &&
             pod.getStatus.getContainerStatuses.stream
               .map[ContainerStateTerminated](cs => cs.getState.getTerminated)
               .anyMatch(t => t != null && t.getExitCode != 0)) {
             PodFailed(pod)
           } else {
-            PodRunning(pod)
+            // Otherwise look for the Spark container
+            val sparkContainerStatusOpt = pod.getStatus.getContainerStatuses.asScala
+              .find(_.getName() == sparkContainerName)
```

Comment:
Sorry, but this seems to fail like the following. Could you check this once more, @holdenk ?
```
containerStatuses=[
ContainerStatus(
  containerID=docker://...,
  image=...,
  imageID=..., 
  lastState=ContainerState(running=null, terminated=null, waiting=null, additionalProperties={}),
  name=pod,
  ready=true,
  restartCount=0,
  started=true,
  state=ContainerState(running=ContainerStateRunning(startedAt=2021-01-03T10:25:24Z, additionalProperties={}), terminated=null, waiting=null, additionalProperties={}),
  additionalProperties={}
)
]
```

Reply:
That makes sense, we'll sometimes not have any status entries for the Spark container. I'll take out the logWarning since this is a more expected state.

Reply:
Thank you, @holdenk .